File: src/main/java/com/ultreon/bubbles/common/logging/old/GameLoggingHandler.java
package com.ultreon.bubbles.common.logging.old;

import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.TimeZone;
import java.util.logging.Handler;
import java.util.logging.Level;
import java.util.logging.LogRecord;

public class GameLoggingHandler extends Handler {
    @Override
    public void publish(LogRecord record) {
        Date date = new Date(record.getMillis());
        DateFormat formatter = new SimpleDateFormat("dd-MM-yyyy HH:mm:ss.SSS XXX");
        formatter.setTimeZone(TimeZone.getDefault());
        String dateFormatted = formatter.format(date);

        if (record.getLevel() == GameLoggingLevels.DEBUG || record.getLevel() == Level.WARNING || record.getLevel() == Level.SEVERE) {
            System.err.println(dateFormatted + " | " + record.getLoggerName() + " | " + record.getLevel().getName() + ": " + record.getMessage());
        } else {
            System.out.println(dateFormatted + " | " + record.getLoggerName() + " | " + record.getLevel().getName() + ": " + record.getMessage());
        }
    }

    @Override
    public void flush() {
        System.out.flush();
    }

    @Override
    public void close() throws SecurityException {
        System.out.close();
    }
}

File: src/main/kotlin/qtech/bubbles/common/logging/old/GameLoggingHandler.kt
package qtech.bubbles.common.logging.old

import java.text.DateFormat
import java.text.SimpleDateFormat
import java.util.*
import java.util.logging.Handler
import java.util.logging.Level
import java.util.logging.LogRecord

class GameLoggingHandler : Handler() {
    override fun publish(record: LogRecord) {
        val date = Date(record.millis)
        val formatter: DateFormat = SimpleDateFormat("dd-MM-yyyy HH:mm:ss.SSS XXX")
        formatter.timeZone = TimeZone.getDefault()
        val dateFormatted = formatter.format(date)
        if (record.level === GameLoggingLevels.DEBUG || record.level === Level.WARNING || record.level === Level.SEVERE) {
            System.err.println(dateFormatted + " | " + record.loggerName + " | " + record.level.name + ": " + record.message)
        } else {
            println(dateFormatted + " | " + record.loggerName + " | " + record.level.name + ": " + record.message)
        }
    }

    override fun flush() {
        System.out.flush()
    }

    @Throws(SecurityException::class)
    override fun close() {
        System.out.close()
    }
}